src/libraries/SqrtPriceMath.sol

```diff
@@ -114,7 +126,13 @@ library SqrtPriceMath {
         pure
         returns (uint160 sqrtQX96)
     {
-        if (sqrtPX96 == 0 || liquidity == 0) revert InvalidPriceOrLiquidity();
+        /// @solidity memory-safe-assembly
+        assembly {
+            if or(iszero(sqrtPX96), iszero(liquidity)) {
```

Comment:
you know me... would love each revert to have a 1 line comment saying
`// equivalent: if (sqrtPX96 == 0 || liquidity == 0) revert InvalidPriceOrLiquidity();`

just to make it easier to read through the code quickly